app/src/main/java/org/mozilla/fenix/perf/Performance.kt

```diff
@@ -4,9 +4,37 @@
 
 package org.mozilla.fenix.perf
 
+import androidx.core.view.doOnPreDraw
+import kotlinx.android.synthetic.main.activity_home.*
+import org.mozilla.fenix.HomeActivity
+
 /**
  * A collection of objects related to app performance.
  */
 object Performance {
     const val TAG = "FenixPerf"
+
+    /**
+     * Instruments cold startup time for use with our internal measuring system, FNPRMS. This may
+     * also appear in Google Play Vitals dashboards.
+     *
+     * This will need to be rewritten if any parts of the UI are changed to be displayed
+     * asynchronously.
+     *
+     * In the current implementation, we only intend to instrument cold startup to the homescreen.
+     * To save implementation time, we ignore the fact that the RecyclerView draws twice if the user
+     * has tabs, collections, etc. open: the "No tabs" placeholder and a tab list. This
+     * instrumentation will only capture the "No tabs" draw.
+     */
+    fun instrumentColdStartupToHomescreenTime(activity: HomeActivity) {
+        // For greater accuracy, we could add an onDrawListener instead of a preDrawListener but:
+        // - single use onDrawListeners are not built-in and it's non-trivial to write one
+        // - the difference is timing is minimal (< 7ms on Pixel 2)
+        // - if we compare against another app using a preDrawListener, it should be comparable
+        //
+        // Unfortunately, this is tightly coupled to the root view of HomeActivity's view hierarchy
+        activity.rootContainer.doOnPreDraw {
+            activity.reportFullyDrawn()
```

Comment:
From my investigation, as best I can tell, the `Displayed` value is calculated based on the time to /complete/ the first draw. So, that might be the source of some of the difference that you mentioned in the notes to this PR.

Reply:
Can you tell me why we are adding the listener to the activity container here and not the recycler view that holds the session controller? I know that we are only concerned about the time that it takes when there are no sessions/tabs/etc open, but to capture even the drawing of those placeholder elements in that controller seems like part of the time that we want to capture in this metric.

I am sure that you know what you are doing which is why I want you to help me understand!

Reply:
> So, that might be the source of some of the difference that you mentioned in the notes to this PR.

Yep, I agree.

> Can you tell me why we are adding the listener to the activity container here and not the recycler view that holds the session controller?

Implementation simplicity. If all of the views are drawn at the same time, we can add a preDraw listener to any of the views and get the same result. However, this is fragile because, if the draw time of the view we add a listener to diverges from the views we care about, those listeners will no get called at the same time.

In this case, I chose to go with the simple implementation instead of the correct one because I figured if we needed `reportFullyDrawn` for real, we'd implement #7781 anyway.

Reply:
Let me know if that doesn't make sense!